src/usr/local/www/pkg_mgr_install.php

```diff
@@ -389,12 +390,25 @@ function waitfor_string_in_file($filename, $string, $timeout) {
 	print_info_box(gettext("Unable to retrieve system versions"), 'danger');
 }
 
+if ($_POST) {
+	$pkgid = str_replace(array("<", ">", ";", "&", "'", '"', '.', '/'), "", htmlspecialchars_decode($_POST['id'], ENT_QUOTES | ENT_HTML401));
+}
+
 if ($_POST['mode'] == 'delete') {
-	$modetxt = gettext("removal");
+	$panel_heading_txt = gettext("Package removal");
+	$pkg_success_txt = sprintf(gettext('<b>%1$s</b> removal successfully completed'), $pkgid);
+	$pkg_fail_txt = sprintf(gettext('<b>%1$s</b> removal failed!'), $pkgid);
+	$pkg_wait_txt = sprintf(gettext('Please wait while the removal of <b>%1$s</b> completes.'), $pkgid);
 } else if (($_POST['mode'] == 'reinstallpkg') || ($_POST['mode'] == 'reinstallall')) {
-	$modetxt = gettext("reinstallation");
+	$panel_heading_txt = gettext("Package reinstallation");
+	$pkg_success_txt = sprintf(gettext('<b>%1$s</b> reinstallation successfully completed'), $pkgid);
+	$pkg_fail_txt = sprintf(gettext('<b>%1$s</b> reinstallation failed!'), $pkgid);
+	$pkg_wait_txt = sprintf(gettext('Please wait while the reinstallation of <b>%1$s</b> completes.'), $pkgid);
 } else {
-	$modetxt = gettext("installation");
+	$panel_heading_txt = gettext("Package installation");
```

Comment:
There is a separate if () lower down that has the special text for updating the base system
gettext("Updating system")
So $panel_heading_txt is never used in that case.